lib/rules/indent.js

```diff
@@ -375,12 +376,25 @@ module.exports = {
                 lastToken = sourceCode.getTokenBefore(lastToken);
             }
 
+            const textBeforeClosingParenthesis = sourceCode.getText(lastToken, lastToken.loc.start.column).slice(0, -1);
+            const indentChar = indentType === "space" ? " " : "\t";
+            const replaceRegExp = new RegExp(indentChar, "g");
+            const textLenWithoutIndentation = textBeforeClosingParenthesis.replace(replaceRegExp, "").length;
             const endIndent = getNodeIndent(lastToken, true);
 
-            if ((endIndent.goodChar !== lastLineIndent || endIndent.badChar !== 0) && isNodeFirstInLine(node)) {
+            /* we consider both of these cases legit:
+                return ((bar === 1 || bar === 2) &&
+                  (z === 3 || z === 4)
+                );
+                return ((bar === 1 || bar === 2) &&
+                  (z === 3 || z === 4));
+              */
+            if ((((endIndent.goodChar !== firstLineIndent && !textLenWithoutIndentation)
+                    || (textLenWithoutIndentation && endIndent.goodChar !== lastLineIndent))
```

Comment:
This line seems redundant to me. `endIndent.goodChar` and `lastLineIndent` are both obtained by calling `getNodeIndent` on the last line of `node`, so I think `endIndent.goodChar !== lastLineIndent` will always be false. I think this check can simply be removed; the important thing is that the node is only reported if `)` is the first token in the line.